Full API Implementation

mrb0nj opened the pull request:
Placeholder PR for full API implementation

jchannon committed:
Post as app apis

jchannon committed:
Added json net attributes

mrb0nj committed:
Use alternate secret

mrb0nj committed:
First stab at restructuring the project for full Api implementation

jchannon committed:
Tidy up to support netstandard2.0 (#76)

mrb0nj committed:
Fix obsolete usages

mrb0nj committed:
Add some missing bits for nuget pack

mrb0nj committed:
Ignore .vscode directory

jchannon commented on build.cake:
```diff
@@ -75,12 +75,12 @@ Task("Pack")
+      Configuration = configuration,
+      NoBuild = true,
+      OutputDirectory = "./artifacts",
+      ArgumentCustomization = args => args.Append($"-p:PackageVersion={gitVersion.NuGetVersionV2}")
```
This can be removed if you update the Version value in the csproj

jchannon replied:
Might also have edit the Version task as that won't be updating AssemblyInfo anymore

mrb0nj committed:
Revised Build Pipeline (#80)

Remove CakeBuild dependency completely
Stop publishing (intermittently) to GitHub Package Repo
Publish to Feedz.io

mrb0nj committed:
Add Feedz.io badge to README

mrb0nj committed:
Only push nuget package if branch is master or commit tagged

mrb0nj committed:
Setup the API Methods in both constructors

mrb0nj committed:
Big ol' Refactor

mrb0nj committed:
ConfigureAwait workaround

mrb0nj committed:
Implement chat.delete

mrb0nj committed:
Rename ChatMessage to PostMessage

mrb0nj committed:
WIP: chat.getPermalink

mrb0nj committed:
Fix GetAsync<T> method to correctly set the query data

mrb0nj committed:
Rename Permalink methods to match API

mrb0nj committed:
Implement chat.postEphemeral method
